packages/dashboard/src/components/queryEditor/advancedSearch/advancedSearch.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import SpaceBetween from '@cloudscape-design/components/space-between';
+import Button from '@cloudscape-design/components/button';
+import FormField from '@cloudscape-design/components/form-field';
+import Input from '@cloudscape-design/components/input';
+import { useState, useMemo, useCallback } from 'react';
+import { ExecuteQueryCommandOutput } from '@aws-sdk/client-iottwinmaker';
+import { createKnowledgeGraphQueryClient } from '@iot-app-kit/react-components/src/components/knowledge-graph/KnowledgeGraphQueries';
+import { TwinMakerKGQueryDataModule } from '@iot-app-kit/source-iottwinmaker';
+import { AdvancedResultsTable } from './advancedResultsTable';
+import { Property } from './model/property';
+
+export interface AdvancedSearchProps {
+  kGDatamodule: TwinMakerKGQueryDataModule;
+  onSelect: (properties: Property[]) => void;
+}
+
+/*
+    The return types are incorrect for Knowledge Graph
+    when selecting an edge instead of a node
+    so we can't properly type here.
+
+    When SiteWise sync is turned on with Knowledge Graph
+    it adds a definition field for the propertyId and displayName
+    and adds propertyName as a field.
+  */
+function executeQueryToProperty(res: any) {
+  const properties: Property[] = [];
+
+  if (res.rows) {
+    for (const row of res.rows) {
+      if (row.rowData) {
+        for (const data of row.rowData) {
+          if (data && data.definition) {
+            properties.push({
+              propertyId: data.definition.configuration.sitewisePropertyId,
+              displayName: data.definition.displayName ? data.definition.displayName : '-',
+              propertyName: data.propertyName,
+            });
+          }
+        }
+      }
+    }
+  }
+
+  return properties;
+}
```

Comment:
Iterating over a 2D array can be expensive. Is it possible we'll run into scalability issues here?

Reply:
I changed this in my latest commit. As we discussed, the new parsing should be O(n) + O(f) = O(n) time complexity where n is the number of matching properties and f is the number of fields we are selecting. This shouldn't result in scalability issues, but we should specifically test the use case of thousands of matching properties to ensure the parsing doesn't cause scalability issues.